### PR TITLE
Validate balaDownloadResponse in pullPackage

### LINKED ISSUE #37229: pullPackage BALA download HTTP response code is not validated
**Description:**
CentralAPIClient->pullPackage method pulls a package from the ballerina central. 

It first calls `[GET] /packages/{orgNamePath}/{packageNamePath}/{versionPath}` to get the url of the location of the package BALA. The response of this request is properly validated.

The next HTTP GET request is to download the BALA using the received URL. The response status code of this, is not validated causing a bad sad error if the request fails.

**Steps to reproduce:**

**Affected Versions:**
Ballerina SL 2201.1.x, 2201.2.x

**OS, DB, other environment details and versions:**

**Related Issues (optional):**
https://github.com/wso2-enterprise/internal-support-ballerina/issues/150

### PR DESCRIPTION
**Pull request by gayaldassanayake**
## Purpose
Handle failures of the BALA download HTTP request in pullPackage method.

Fixes #37229

## Approach
Add validation that checks the response status code and throws errors.

## Remarks
Partially addresses https://github.com/wso2-enterprise/internal-support-ballerina/issues/150

### CODE REVIEW

**Review by azinneera — DISMISSED** [review]

File: cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java [hunk 1]
```diff
@@ -896,7 +905,16 @@ protected void handleResponseErrors(Response response, String msg) throws Centra
                     response.code() == HTTP_UNAVAILABLE) {
                 Error error = new Gson().fromJson(body.get().string(), Error.class);
                 if (error.getMessage() != null && !"".equals(error.getMessage())) {
-                    throw new CentralClientException(msg + "' reason:" + error.getMessage());
+                    throw new CentralClientException(msg + " reason:" + error.getMessage());
+                }
+            }
+
+            // If invalid/no response received at the gateway server
+            if (response.code() == HTTP_BAD_GATEWAY ||
```
Comment (Dilhasha) [inline cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java:913]: What happens if an HTTP response code other than the handled ones are received?

File: cli/central-client/src/test/java/org/ballerinalang/central/client/TestCentralApiClient.java [hunk 2]
```diff
@@ -208,6 +227,93 @@ public void testPullNonExistingPackage() throws IOException, CentralClientExcept
...
+                getResponseBody(responseString)};
+        data[2] = new Object[]{HttpURLConnection.HTTP_BAD_GATEWAY, remoteRepositoryErrorMessage,
+                getResponseBody(responseString)};
+        data[3] = new Object[]{HttpURLConnection.HTTP_UNAVAILABLE, remoteRepositoryErrorMessage,
+                getResponseBody(responseString)};
+        data[4] = new Object[]{HttpURLConnection.HTTP_GATEWAY_TIMEOUT, remoteRepositoryErrorMessage,
+                getResponseBody(responseString)};
+
+        // test BALA download with no response body
+        data[5] = new Object[]{HttpURLConnection.HTTP_UNAVAILABLE, errorWithNoResponseBodyMessage, null};
```
Comment (Dilhasha) [inline cli/central-client/src/test/java/org/ballerinalang/central/client/TestCentralApiClient.java:253]: Shall we add a test case to validate the behavior when an unexpected error code is received such as `HTTP_FORBIDDEN`?

File: cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java (hunk 1 above)
Comment (gayaldassanayake) [reply cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java:913]: It throws an CentralClientException with the provided error message. 

Line 922. https://github.com/gayaldassanayake/ballerina-lang/blob/eb0f3abb4f6268f95e1ad420eee5f9b99e0cf752/cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java#L922.

Will add a test case representing this scenario.

File: cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java [hunk 3]
```diff
@@ -896,7 +905,16 @@ protected void handleResponseErrors(Response response, String msg) throws Centra
...
+                    response.code() == HTTP_GATEWAY_TIMEOUT) {
```
Comment (Dilhasha) [inline cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java:914]: Can't we combine the 2 conditions at https://github.com/gayaldassanayake/ballerina-lang/blob/eb0f3abb4f6268f95e1ad420eee5f9b99e0cf752/cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java#L904 and here? Does the same processing right?

Comment (gayaldassanayake) [reply cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java:914]: Yes. Will update 👍

**Review by Dilhasha — APPROVED** [review]

**Pull request merged**